src/draw/external_image.rs

```diff
@@ -0,0 +1,56 @@
+use webrender::api::*;
+
+use render::RenderBuilder;
+use widget::draw::Draw;
+use resources::resources;
+use geometry::{Rect, RectExt, Size, SizeExt};
+
+pub struct ExternalImageState {
+    pub name: String,
+    pub data: ImageData,
+}
+
+impl ExternalImageState {
+    pub fn new(name: &str, texture_id: u64) -> ExternalImageState {
+        let data = ImageData::External(ExternalImageData {
+            id: ExternalImageId(texture_id),
+            channel_index: 0,
+            image_type: ExternalImageType::Texture2DHandle,
+        });
+        let descriptor = ImageDescriptor::new(0, 0, ImageFormat::RGB8, true);
+        resources().put_image(name, data.clone(), descriptor);
```

Comment:
Instead of reusing the `images` map, and using user specified names that could potentially collide with each other or with file names, maybe it should use a generated id as a key? Not a huge deal either way, I can always change it later

Reply:
Yeah I'd like to not have any names at all.

Honestly I don't entirely understand the whole centralized `resources` thing…

Reply:
Yeah it might not even need to be centralized and global anymore, it was added when the architecture was completely different, long before limn used webrender. Don't worry about it, I'll do some refactoring on it later on.

Reply:
Wow, didn't know limn has a long history!